bitrepository-client/src/main/java/org/bitrepository/client/conversation/GeneralConversationState.java

```diff
@@ -71,8 +74,9 @@ protected GeneralConversationState(Collection<String> expectedContributors) {
     public void start() {
         try {
             if (!responseStatus.getOutstandingComponents().isEmpty()) {
-                if (getTimeoutValue() > 0) {
-                    scheduledTimeout = timer.schedule(new TimeoutHandler(), getTimeoutValue(), TimeUnit.MILLISECONDS);
+                if (getTimeoutValue().compareTo(Duration.ZERO) > 0) { // From Java 18 use: getTimeoutValue().isPositive()
```

Comment:
Add the text 'TODO' to this comment so it is easier to notice later on, especially since we will most likely upgrade to Java 18 soon-ish

Reply:
Thanks. Fixed.